src/sentry/middleware/devtoolbar.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+from django.http import HttpRequest, HttpResponse
+
+from sentry import analytics
+from sentry.utils.http import origin_from_request
+from sentry.utils.http import query_string as get_query_string
+from sentry.utils.urls import parse_id_or_slug_param
+
+logger = logging.getLogger(__name__)
+
+
+class DevToolbarAnalyticsMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        try:
+            response = self.get_response(request)
+        except Exception:
+            response = HttpResponse(status=500)
+            raise  # delegate to our API exception handlers
+        finally:
+            _record_api_request(request, response)
+
+        return response
+
+
+def _record_api_request(request: HttpRequest, response: HttpResponse) -> None:
```

Comment:
can we put an option around this for a quick rollback incase something goes wrong with this?